pkg/didcomm/protocol/didexchange/states.go

```diff
@@ -30,7 +36,7 @@ type state interface {
 	CanTransitionTo(next state) bool
 	// Executes this state, returning a followup state to be immediately executed as well.
 	// The 'noOp' state should be returned if the state has no followup.
-	Execute(msg dispatcher.DIDCommMsg) (followup state, err error)
+	Execute(msg dispatcher.DIDCommMsg, ctx context) (followup state, err error)
```

Comment:
what I see only one method requires context, better to use smt like this
`Execute(msg dispatcher.DIDCommMsg, ... opts)`